File: 1162. As Far from Land as Possible/Solution.java
class Solution {
    public int maxDistance(int[][] grid) {
        int[][] move = new int[][] { { 0, 1 }, { 0, -1 }, { 1, 0 }, { -1, 0 } };
        int n = grid.length, dist = 0;
        Queue<int[]> q = new LinkedList<>();
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (grid[i][j] == 1) q.offer(new int[] { i, j });
            }
        }
        if (q.size() == 0 || q.size() == n * n) return -1;
        while (!q.isEmpty()) {
            int l = q.size();
            for (int i = 0; i < l; i++) {
                int[] curr = q.poll();
                for (int[] m : move) {
                    if (curr[0] + m[0] < 0 || curr[0] + m[0] >= n || curr[1] + m[1] < 0 || curr[1] + m[1] >= n) continue;
                    if (grid[curr[0] + m[0]][curr[1] + m[1]] == 1) continue;
                    grid[curr[0] + m[0]][curr[1] + m[1]] = 1;
                    q.offer(new int[] { curr[0] + m[0], curr[1] + m[1] });
                }
            }
            dist++;
        }
        return dist - 1;
    }
}

File: 1162. As Far from Land as Possible/Solution.py
class Solution:
    def maxDistance(self, grid: List[List[int]]) -> int:
        move = [ [0, 1], [0, -1], [1, 0], [-1, 0]]
        n, dist = len(grid), 0
        q = []
        for i in range(n):
            for j in range(n):
                if grid[i][j] == 1: q.append([i, j])
        if not q or len(q) == n * n: return -1
        while q:
            l = len(q)
            for i in range(l):
                x, y = q.pop(0)
                for a, b in move:
                    if x + a < 0 or x + a >= n or y + b < 0 or y + b >= n: continue
                    if grid[x + a][y + b] == 1: continue
                    grid[x + a][y + b] = 1
                    q.append([x + a, y + b])
            dist += 1
        return dist - 1